code/modules/mob/living/carbon/human/human.dm

```diff
@@ -150,15 +147,19 @@
 				Paralyse(10)
 
 	take_overall_damage(b_loss,f_loss)
-
+	//attempt to dismember bodyparts
+	for(var/X in bodyparts)
+		var/obj/item/bodypart/BP = X
+		if(prob(50/severity) && BP.body_zone != "head" && BP.body_zone != "chest")
+			BP.dismember()

```

Comment:
> LIkewise, bomb armor should probably reduce or eliminate the chances of this happening.

The comment where I also requested this was hidden by a different change in code. Phil stated that he will handle this in a separate PR.